python/lsst/resources/tests.py

```diff
@@ -28,6 +28,19 @@
 TESTDIR = os.path.abspath(os.path.dirname(__file__))
 
 
+def prepare_testing(testcase: unittest.TestCase) -> None:
```

Comment:
Since this is specifically for use by S3 tests only I would be inclined to put this in `s3utils` and not the generic `tests.py`. Pondering the name of the function, I think I would prefer something like `clean_test_environment`. You will need to add to `__all__` when you move it.